src/App.js

```diff
@@ -1,37 +1,95 @@
-import "./App.css";
-import { useState } from "react";
+import './App.css';
+import './styles/variables.css';
+import './styles/core.css';
+
+import { useState } from 'react';
+import { ReactQueryDevtools } from 'react-query/devtools';
+
+import { QueryClient, QueryClientProvider } from 'react-query';
+
+import Layout from './Components/Layout';
+
+import TodoApp from './Components/Todo';
+import Login from './Components/Auth/Login';
+import Signup from './Components/Auth/Signup';
+
+import { TodoProvider } from './Context';
+
+import {
+  clearLocalStorage,
+  isAuthenticated,
+  JWT_TOKEN,
+} from './utils/commonUtils';
+
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      refetchOnWindowFocus: false,
+    },
+  },
+});
 
 function App() {
-  const [todos] = useState([
-    "Add a button that allows users to add an item in todo list",
-    "Implement context",
-  ]);
+  const [showSignup, setShowSignup] = useState(false);
+  const [showProfileEdit, setShowProfileEdit] = useState(false);
+
+  const [authenticated, setAuthenticated] = useState(false);
+
+  const showProfileEditComponent = () => {
+    setShowSignup(false);
+    setShowProfileEdit(true);
+  };
+
+  const hideProfileEditComponent = () => {
+    setShowSignup(false);
+    setShowProfileEdit(false);
+  };
+
+  const handleSignout = () => {
+    clearLocalStorage(JWT_TOKEN);
+    setAuthenticated(false);
+    window.location.reload();
+  };
+
   return (
     <div className="App">
-      <section>
-        <h4>Your assignment</h4>
-        <p>You cannot use any library to do this task</p>
-        <p>A todo item should not be empty</p>
-        <p>A todo item cannot contain word: "Postpone" in any format</p>
-        <ul className="assignment-tasks">
-          <li>Allow users to add an item in to do list</li>
-          <li>
-            Allow users to remove an individual item in to do list
-          </li>
-          <li>
-            Separate todoitems listing component into a different component
-          </li>
-          <li>Bonus points if you can use Context to implement it</li>
-        </ul>
-      </section>
-      <section>
-        <h3>Todos</h3>
-        <ul className="todos">
-          {todos.map((todoItem) => (
-            <li key={todoItem}>{todoItem}</li>
-          ))}
-        </ul>
-      </section>
+      <TodoProvider
+        value={{
+          queryClient,
+          handleSignout,
+          showProfileEdit,
+          showProfileEditComponent,
+          hideProfileEditComponent,
+        }}
+      >
+        <QueryClientProvider client={queryClient}>
+          {showProfileEdit ? (
```

Comment:
Ternary inside a ternary is a bad practice.

Reply:
Refactored the code. Removed the use of ternary operators. Do have a look.
Although it would be a lot cleaner with routing, the use of any library is not allowed as per the task rules.